src/main/java/practice/StreamPractice.java

```diff
@@ -1,80 +1,71 @@
-        return 0;
+        return numbers.stream()
+                .flatMap(n -> Arrays.stream(n.split("\\s*,\\s*")))
+                .mapToInt(Integer::parseInt)
+                .filter(n -> n % 2 == 0)
+                .min()
+                .orElseThrow(() ->
+                        new RuntimeException("Can't get min value from list: " + numbers));
     }
 
-    /**
-     * Given a List of Integer numbers,
-     * return the average of all odd numbers from the list or throw NoSuchElementException.
-     * But before that subtract 1 from each element on an odd position (having the odd index).
-     */
-    public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+    public double getOddNumsAverage(List<Integer> numbers) {
+        return IntStream.range(0, numbers.size()).map(n -> {
+            int value = numbers.get(n);
+            if (n % 2 == 1) {
+                value = value - 1;
+            }
+            return value;
+        })
+                .filter(n -> n % 2 == 1)
+                .average()
+                .getAsDouble();
     }
 
-    /**
-     * Given a List of `Person` instances (having `name`, `age` and `sex` fields),
-     * for example, `Arrays.asList( new Person(«Victor», 16, Sex.MAN),
-     * new Person(«Helen», 42, Sex.WOMAN))`,
-     * select from the List only men whose age is from `fromAge` to `toAge` inclusively.
-     * <p>
-     * Example: select men who can be recruited to army (from 18 to 27 years old inclusively).
-     */
     public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getAge() >= fromAge
+                        && p.getAge() <= toAge
+                        && p.getSex() == Person.Sex.MAN)
+                .collect(Collectors.toList());
     }
 
-    /**
-     * Given a List of `Person` instances (having `name`, `age` and `sex` fields),
-     * for example, `Arrays.asList( new Person(«Victor», 16, Sex.MAN),
-     * new Person(«Helen», 42, Sex.WOMAN))`,
-     * select from the List only people whose age is from `fromAge` and to `maleToAge` (for men)
-     * or to `femaleToAge` (for women) inclusively.
-     * <p>
-     * Example: select people of working age
-     * (from 18 y.o. and to 60 y.o. for men and to 55 y.o. for women inclusively).
-     */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> (p.getAge() >= fromAge
+                        && p.getAge() <= maleToAge && p.getSex() == Person.Sex.MAN)
+                        || (p.getAge() >= fromAge && p.getAge() <= femaleToAge
+                        && p.getSex() == Person.Sex.WOMAN))
+                .collect(Collectors.toList());
```

Comment:
You can use this instead:

```suggestion
                .toList()
```